### PR TITLE
Bug/1507 ensure activating deactivating module refreshes auth state

### PR DESCRIPTION
**Pull request by jqlee85**
## Summary

Addresses issue #1507

## Relevant technical choices

Adds yield in `activateModule` and `deactivateModule` that returns `{ payload: {}, type: REFETCH_AUTHENTICATION }`.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by felixarntz** [comment]
@jqlee85 Quick update, I think adding a `timestamp` query parameter to `refreshAuthentication` is all that's needed here to make this work. Then #1611 (IB has more context) is about fixing this throughout.

**Comment by jqlee85** [comment]
@felixarntz sounds good. It looks like the `refreshAuthentication` utility already passes a unique timestamp each time:
https://github.com/google/site-kit-wp/blob/af5b679c11ba92b266f1a659f9b840823d8173b7/assets/js/util/refresh-authentication.js#L33

Is there something else that needs to be done then or does that resolve the issue?

**Comment by felixarntz** [comment]
@jqlee85 The `refreshAuthentication` utility function isn't used by any of the new code though, it's only legacy code. So we'll need to implement that same timestamp addition to the `REFETCH_AUTHENTICATION` store action. Probably easiest to duplicate the logic that is in the `FETCH_AUTHENTICATION` control and add the timestamp there.

**Comment by jqlee85** [comment]
@felixarntz ah yes, I see. I've made that update. Let me know if there are any further issues.

File: assets/js/googlesitekit/modules/datastore/modules.js [hunk 1]
```diff
@@ -197,6 +210,9 @@ export const controls = {
 	[ FETCH_MODULES ]: () => {
 		return API.get( 'core', 'modules', 'list', null, { useCache: false } );
 	},
+	[ REFETCH_AUTHENICATION ]: () => {
+		return API.get( 'core', 'user', 'authentication', null, { useCache: false, timestamp: Date.now() } );
```
Comment (felixarntz) [inline assets/js/googlesitekit/modules/datastore/modules.js:214]: I'm not sure this is right. Looking at `API.get`, the `timestamp` argument here wouldn't currently do anything, it needs to be in the `data` parameter (which is currently `null`) in order to affect the request URL.

Comment (jqlee85) [reply assets/js/googlesitekit/modules/datastore/modules.js]: @felixarntz Ah yes, I mistook the arguments there, thanks. I've moved the timestamp into the `data` arg, and added an `expect()` in the tests to make sure `timestamp` is included as a param for those GET requests to the auth endpoint. I also checked this locally, and confirmed the timestamp shows up in the GET request during the module deactivation. Let me know if there's anything else to button up here.

**Pull request merged**